pkg/windows/windows.go

```diff
@@ -591,9 +597,32 @@ func (vm *windows) createDirectories() error {
 // removeDirectories removes all directories created as part of the configuration process
 func (vm *windows) removeDirectories() error {
 	vm.log.Info("removing directories")
+	existingFiles, err := getFilesToTransfer()
+	if err != nil {
+		return errors.Wrapf(err, "error getting list of files transferred to the instance")
+	}
+	for src, dest := range existingFiles {
+		filePath := dest + "\\" + filepath.Base(src.Path)
+		err = wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
+			if out, err := vm.Run(rmFileCmd(filePath), true); err != nil {
+				vm.log.V(1).Error(err, "error removing file", "file", filePath, "out", out)
+				return false, nil
+			}
+			out, err := vm.Run("Test-Path "+filePath, true)
+			if err != nil {
+				vm.log.V(1).Error(err, "error checking if file exists", "file", filePath,
+					"out", out)
+				return false, nil
+			}
+			return !(strings.TrimSpace(out) == "True"), nil
+		})
+		if err != nil {
+			return errors.Wrapf(err, "unable to remove file: %s", filePath)
+		}
+	}
 	for _, dir := range RequiredDirectories {
 		if out, err := vm.Run(rmDirCmd(dir), true); err != nil {
-			return fmt.Errorf("unable to remove directory %s, out: %s, err: %s", dir, out, err)
+			vm.log.V(1).Error(err, "unable to remove directory", "dir", dir, "out", out)
```

Comment:
@saifshaikh48 we want to make directory removals best effort so we are deleting files one by one. There is no motivation to remove files sequentially otherwise.